apiserver/common/unitstatus_test.go

```diff
@@ -95,6 +95,44 @@ func (s *UnitStatusSuite) TestCAASLost(c *gc.C) {
 	s.checkLost(c)
 }
 
+func (s *UnitStatusSuite) TestLostTerminated(c *gc.C) {
+	s.unit.status.Status = status.Terminated
+	s.unit.status.Message = ""
+
+	s.ctx.Presence = agentDown(s.unit.Tag().String())
+
+	agent, workload := s.ctx.UnitStatus(s.unit)
+	c.Check(agent.Status, jc.DeepEquals, status.StatusInfo{
+		Status:  status.Lost,
+		Message: "agent is not communicating with the server",
+	})
+	c.Check(agent.Err, jc.ErrorIsNil)
+	c.Check(workload.Status, jc.DeepEquals, status.StatusInfo{
+		Status:  status.Terminated,
+		Message: "",
```

Comment:
shouldn't the message here be `"unit stopped by the cloud"` ?

Reply:
no that's not necessary, in the example I gave for QA that message is set by the controller. Here I simply test that if the application is terminated with an empty message, then the `UnitStatus()` function does not overwrite this (empty) message.